client/src/controllers/SlugController.ts

```diff
@@ -0,0 +1,28 @@
+import { Controller } from '@hotwired/stimulus';
+import { cleanForSlug } from '../utils/text';
+
+/**
+ * Adds ability for the input element value to be slugified when out of focus or blurred.
+ *
+ * @example
+ * <input
+ * type="text"
+ * name="slug"
+ * value="slug testing page"
+ * maxlength="255"
+ * required=""
+ * id="id_slug">
+ */
+export class SlugController extends Controller<HTMLInputElement> {
+  static values = {
+    allowUnicode: { default: false, type: Boolean },
+  };
+
+  declare allowUnicodeValue: boolean;
+
+  slugify() {
+    this.element.value = cleanForSlug(this.element.value, false, {
+      unicodeSlugsEnabled: (this.allowUnicodeValue = true),
```

Comment:
```suggestion
      unicodeSlugsEnabled: this.allowUnicodeValue,
```

This line has a bug in it, you are declaring the value here accidentally (updating it), undoing this change appears to cause some other issues to debug but this line should be just reading out the value not setting it.

Reply:
If we aren't setting the value here, does that mean we're using the valueChanged method of Stimulus controller? But I tried that and it wasn't working properly

```

slugify() {
    this.allowUnicodeValue = true;
  }

  allowUnicodeValueChanged(isAllowUnicode: Boolean) {
    const inputValue = this.element.value.trim();

    if(isAllowUnicode) {
      this.element.value = cleanForSlug(inputValue, false, {
        unicodeSlugsEnabled: (this.allowUnicodeValue),
      });
    }
  }
  ```
  
 returns `"    slug testing on edit page "`

Reply:
I think there might be some confusion here, we do not need to do anything when the 'setting' allow unicode value changes, we only need to read the value when we do the actual update via the `slugify` method.